wasm/src/programs/macros.rs

```diff
@@ -30,6 +30,82 @@ macro_rules! process_inputs {
     }};
 }
 
+#[macro_export]
+macro_rules! authorize {
+    (
+        $process:expr,
+        $inputs:expr,
+        $program_string:expr,
+        $function_id_string:expr,
+        $private_key:expr,
+        $rng:expr
+    ) => {{
+        log("Loading program");
+        let program =
+            ProgramNative::from_str($program_string).map_err(|_| "The program ID provided was invalid".to_string())?;
+        log("Loading function");
+        let function_name = IdentifierNative::from_str($function_id_string)
+            .map_err(|_| "The function name provided was invalid".to_string())?;
+
+        let program_id = program.id().to_string();
+
+        if program_id != "credits.aleo" {
+            if !$process.contains_program(program.id()) {
+                log("Adding program to the process");
+                $process.add_program(&program).map_err(|e| e.to_string())?;
```

Comment:
Just sanity checking: authorisation does *not* require adding the imports to the process before the parent call?